src/js/ripple/serializedtypes.js

```diff
@@ -354,10 +335,13 @@ var STAmount = exports.Amount = new SerializedType({
         }
 
         // Next eight bits: offset/exponent
-        hi |= ((97 + amount._offset) & 0xff) << 22;
+        hi |= ((97 + offset) & 0xff) << 22;
+
         // Remaining 54 bits: mantissa
-        hi |= amount._value.shiftRight(32).intValue() & 0x3fffff;
-        lo = amount._value.intValue() & 0xffffffff;
+        var mantissaDecimal = utils.getMantissaDecimalString(value.abs());
+        var mantissaHex = (new BigNumber(mantissaDecimal)).toString(16);
```

Comment:
Tried that, but it doesn't work because sjcl doesn't parse decimal (or at least I couldn't find anything about it in the docs).